solc/CommandLineInterface.cpp

```diff
@@ -1762,16 +1752,36 @@ void CommandLineInterface::handleAst()
 	else
 	{
 		sout() << "JSON AST (compact format):" << endl << endl;
-		for (auto const& sourceCode: m_sourceCodes)
+		for (auto const& sourceCode: m_fileReader.sourceCodes())
 		{
 			sout() << endl << "======= " << sourceCode.first << " =======" << endl;
 			ASTJsonConverter(m_compiler->state(), m_compiler->sourceIndices()).print(sout(), m_compiler->ast(sourceCode.first));
 		}
 	}
 }
 
+bool CommandLineInterface::serveLSP()
+{
+	auto const lspPortStr = m_args.count("lsp-port") != 0 ? m_args.at("lsp-port").as<string>() : "0"s;
+
+	auto const traceLogger = [](string_view _msg) {
+		fprintf(stderr, "%s\n", string(_msg).c_str());
+	};
+
+	unique_ptr<::lsp::Transport> transport;
+	if (lspPortStr.empty())
+		transport = make_unique<::lsp::JSONTransport>(traceLogger);
+	else
+		transport = make_unique<::lsp::TcpTransport>(stoi(lspPortStr), traceLogger);
```

Comment:
Is this just for debugging now, or do we need to link against networking code?

Reply:
I am absolutely find in throwing the TcpTransport part away right before merge. But for now, it's most convenient when debugging (It could also be #ifdef'ed out in release builds)